### PR TITLE
First commit

### PR DESCRIPTION
**Pull request by ChristianSantillanS**
Functionality to be able to change the user language from the top nav and the mobile menu
English:
image
Spanish:
image

### CODE REVIEW

File: components/Navbar/NavbarUserMenu.tsx [hunk 1]
```diff
@@ -17,11 +17,24 @@ const FirebaseAuthLoader = dynamic(
...
+    lang = 'en';
+  } else {
+    location.href = '/es';
+    lang = 'es';
+  }
+}
 const NavbarUserMenu = () => {
+  const { t } = useTranslation('common');
   const { lang } = useTranslation();
+  const Profile = t('Profile');
```
Comment (interactivelabs) [inline components/Navbar/NavbarUserMenu.tsx:32]: I think inlining this on the body of the jsx it's better in this case because we really don't gain anything from this assignment and it does makes this section more verbose :)

File: components/Navbar/NavbarUserMenu.tsx [hunk 2]
```diff
@@ -17,11 +17,24 @@ const FirebaseAuthLoader = dynamic(
   { ssr: false }
 );
 
+function handleClick(this: any, lang: string) {
+  if (lang == 'es') {
+    location.href = '/en';
```
Comment (interactivelabs) [inline components/Navbar/NavbarUserMenu.tsx:22]: I think there is a way to do this with the Next.js router not the browser object

**Review by interactivelabs — COMMENTED** [review]
Also where is the part of saving the selection on a cookie?

File: providers/CommerceProvider.tsx [hunk 3]
```diff
@@ -1,68 +1,48 @@
...
+
+export function usePersistLocaleCookie() {
+  const { locale, defaultLocale } = useRouter();
+  useEffect(persistLocaleCookie, [locale, defaultLocale]);
+  function persistLocaleCookie() {
+    if (locale !== defaultLocale) {
+      const date = new Date();
+      const expireMs = 100 * 24 * 60 * 60 * 100;
+      date.setTime(date.getTime() + expireMs);
+      document.cookie = `NEXT_LOCALE=${locale};expires=${date.toUTCString()};path=/`;
```
Comment (interactivelabs) [inline providers/CommerceProvider.tsx:45]: please use the nookies package: https://github.com/maticzav/nookies#readme

File: providers/AppContexts.tsx [hunk 4]
```diff
@@ -5,15 +5,19 @@ import { getCookieIdToken } from '@lib/firebaseAuth/utils';
...
   children: ReactNode;
   pageProps: BaseObject;
 }
 
 const AppContexts = ({ children, pageProps }: IAppContexts) => {
+  // const { locale } = useCommerce();
   const apolloClient = useApollo(pageProps, {
     graphQLUrl: getGraphqlURL(),
     getToken: getCookieIdToken,
+    //locale, //Error
```
Comment (interactivelabs) [inline providers/AppContexts.tsx:20]: you need to update: https://github.com/interactivelabsmx/mundoballoon-commerce/blob/main/lib/apollo/apolloClient.ts#L18

To be able to accept this new parameter and pass it down to the client headers.

File: providers/CommerceProvider.tsx [hunk 5]
```diff
@@ -1,68 +1,48 @@
...
 
 interface ICommerce {
   children: ReactNode;
-  options?: Partial<ICommerceConfig>;
 }
 
-export function CommerceProvider({ children, options }: ICommerce) {
-  const value = useMemo(() => ({ ...CommerceConfig, options }), [options]);
+export function CommerceProvider({ children }: ICommerce) {
+  const [locale, setLocale] = useState<string>(defaultLocale);
```
Comment (interactivelabs) [inline providers/CommerceProvider.tsx:28]: Here check if the cookie is set read from it to set te default value.

File: components/Navbar/NavbarUserMenu.tsx [hunk 6]
```diff
@@ -18,10 +19,21 @@ const FirebaseAuthLoader = dynamic(
...
+  let langs = '';
+  const { t } = useTranslation('common');
   const { lang } = useTranslation();
+  const Profile = t('Profile');
   const { user, logout } = useAuth();
   const [openAuth, setOpenAuth] = useState(false);
   const onClick = () => setOpenAuth(true);
+  const Lang = t('Lang');
+  const Sign_Out = t('Sign Out');
+  if (lang == 'es') {
```
Comment (interactivelabs) [inline components/Navbar/NavbarUserMenu.tsx:31]: what is langs used for?

File: components/Products/ProductLists/ProductListHero/ProductListHero.tsx [hunk 7]
```diff
@@ -15,7 +15,7 @@ const ProductListHero = ({ productDictionary }: IProductListHero) => {
           <div className="py-16 sm:py-24 lg:max-w-7xl lg:mx-auto lg:px-8">
             <div className="py-8 px-4 flex items-center justify-between sm:px-6 lg:px-0">
               <h2 className="text-2xl font-extrabold tracking-tight text-gray-900">
-                {productsItem.key}
+                {t(productsItem.key)}
```
Comment (interactivelabs) [inline components/Products/ProductLists/ProductListHero/ProductListHero.tsx:18]: This is coming from the database, this shouldn't be translated like this, if you want to change the text change it on the DB please :) thanks

File: components/User/Auth/FirebaseAuth.tsx [hunk 8]
```diff
@@ -63,7 +66,7 @@ const FirebaseAuth = () => {
           </div>
           <div className="relative flex justify-center text-sm">
             <span className="px-2 bg-gray-100 text-gray-500">
-              Or continue with
+              {Or_continue_with}
```
Comment (interactivelabs) [inline components/User/Auth/FirebaseAuth.tsx:69]: you can in-line {t('Or continue with');} no need to assign a variable like that

File: locales/es/common.json [hunk 9]
```diff
@@ -1,3 +1,11 @@
 {
-  "see_everything": "Ver todas las opciones"
+  "see_everything": "Ver todas las opciones",
+  "Featured": "Destacados",
+  "logout": "Cerrar Sesión",
+  "Sign in": "Iniciar sesión",
```
Comment (interactivelabs) [inline locales/es/common.json:5]: This keys on the translation should also be added to English.

File: components/User/Auth/FirebaseEmailAuth.tsx [hunk 10]
```diff
@@ -10,6 +10,8 @@ import LoadingText from '@components/UI/loading/LoadingText';
 import unifiedEmailPasswordAuth from '@lib/firebaseAuth/unifiedEmailPasswordAuth';
 import { useAuth } from '@providers/AuthProvider';
 
+//const { t } = useTranslation('common');
+const password = 'Password';
```
Comment (interactivelabs) [inline components/User/Auth/FirebaseEmailAuth.tsx:14]: revert this change as it was before, 
label="Password"

File: pages/index.tsx [hunk 11]
```diff
@@ -17,6 +16,6 @@ const Index = () => {
   );
 };
 
-export const getServerSideProps = getServerSidePreFetch({ Page: Index });
+//export const getServerSideProps = getServerSidePreFetch({ Page: Index });
```
Comment (interactivelabs) [inline pages/index.tsx:19]: un comment this like the file does needs to prefetch

File: providers/AuthProvider.tsx [hunk 12]
```diff
@@ -78,5 +79,21 @@ export function AuthProvider({ children }: IAuthProvider) {
     </AuthContext.Provider>
   );
 }
+export interface ICommerceProvider {
```
Comment (interactivelabs) [inline providers/AuthProvider.tsx:82]: why is this on this file? this should be on CommerceProvider.

File: providers/AuthProvider.tsx [hunk 13]
```diff
@@ -78,5 +79,21 @@ export function AuthProvider({ children }: IAuthProvider) {
...
+  locale: string;
+  setLocale: (newLang: string) => void;
+}
+export function LocaleCookie(): void {
```
Comment (interactivelabs) [inline providers/AuthProvider.tsx:86]: This is never called and this should also be on the CommerceProvider

File: providers/CommerceProvider.tsx [hunk 14]
```diff
@@ -1,65 +1,31 @@
...
-  options?: Partial<ICommerceConfig>;
+  lang: string;
 }
 
-export function CommerceProvider({ children, options }: ICommerce) {
-  const value = useMemo(() => ({ ...CommerceConfig, options }), [options]);
+export function CommerceProvider({ children, lang }: ICommerce) {
+  const locale = lang;
+  const setLocale = (newlang: string) => {
+    setLanguage(newlang);
```
Comment (interactivelabs) [inline providers/CommerceProvider.tsx:23]: you should call the function that sets the cookie here.

File: providers/AuthProvider.tsx [hunk 15]
```diff
@@ -78,5 +79,21 @@ export function AuthProvider({ children }: IAuthProvider) {
...
+  const { lang } = useTranslation('common');
+  const { locale, defaultLocale } = useRouter();
+  useEffect(persistLocaleCookie, [locale, defaultLocale]);
+  function persistLocaleCookie() {
+    if (locale != defaultLocale) {
+      setCookie({}, FI, 'value', {
```
Comment (interactivelabs) [inline providers/AuthProvider.tsx:92]: the FI cookie is already used for auth, please use a new name.

Comment (ChristianSantillanS) [reply providers/AuthProvider.tsx:92]: Can I call the cookie 'FII'?

File: lib/firebaseAuth/utils.ts [hunk 16]
```diff
@@ -10,12 +10,25 @@ export const FI_COOKIE_OPTIONS = {
...
+export const FII_TTL = 60 * 60;
+export const FII_COOKIE_OPTIONS = {
+  path: '/',
+  maxAge: FII_TTL,
+  sameSite: true,
+  secure: true,
+};
+
 export const FI = 'fi';
+export const FII = 'fii';
```
Comment (interactivelabs) [inline lib/firebaseAuth/utils.ts:22]: I named the other cookie FI for FirebaseInformation... 🤔  I think we can use a better name for this cookie, something like UL for UserLanguage :)

File: components/Navbar/NavbarUserMenu.tsx [hunk 17]
```diff
@@ -18,10 +19,13 @@ const FirebaseAuthLoader = dynamic(
...
 
 const NavbarUserMenu = () => {
-  const { lang } = useTranslation();
+  const { t, lang } = useTranslation('common');
+  const Profile = t('Profile');
   const { user, logout } = useAuth();
   const [openAuth, setOpenAuth] = useState(false);
   const onClick = () => setOpenAuth(true);
+  const Lang = t('Lang');
+  const Sign_Out = t('Sign Out');
```
Comment (interactivelabs) [inline components/Navbar/NavbarUserMenu.tsx:28]: please inline this :) not required to have another var per translation t('Profile'), t('Lang') t('Sign Out') can just be placed on the jsx it self (inline it).

File: lib/firebaseAuth/utils.ts [hunk 18]
```diff
@@ -38,3 +51,10 @@ export const checkForExpiredCookieToken = (refreshIdToken: () => void) => {
     if (!cookie) refreshIdToken();
   }, FI_TTL * 1000);
 };
+
+export const ckeckForExpiredCookieLanguage = (refreshLanguage: () => void) => {
```
Comment (interactivelabs) [inline lib/firebaseAuth/utils.ts:55]: Also as I mentioned move all this language functions to it's own file or in the CommerceContext, this file 'lib/firebaseAuth/utils.ts' it's only for auth related functions :)

**Review by interactivelabs — DISMISSED** [review]

**Review by ChristianSantillanS — COMMENTED** [review]
Language and cookie

**Review by interactivelabs — APPROVED** [review]
